add routing code to master

Vadangowda opened the pull request.

PJMODOS commented:
This pull request has no description, fails in CI and seems to contain code that is licensed under "Amazon Software License"  rather then "ISC License" that we use for the pgbouncer code.

Can you please explain what you are trying to achieve here?

petere commented:
I suspect it's from https://github.com/awslabs/pgbouncer-rr-patch, but it clearly needs to be explained better first of all.

Vadangowda commented:
Hi All,

i added routing based on the schema name in the MR.  To achieve this
initially  i integrated pgbouncer-rr, but since that is written in python
and was loaded on every query it was not optimal.
so i rewrote the whole thing using C. Let me clean up the whole thing and
remove all the license coming from aws.

Regards
Vadan

On Tue, Feb 25, 2020 at 12:54 PM Peter Eisentraut <notifications@github.com>
wrote:

> I suspect it's from https://github.com/awslabs/pgbouncer-rr-patch, but it
> clearly needs to be explained better first of all.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pgbouncer/pgbouncer/pull/466?email_source=notifications&email_token=AN4AB7NJJIXXKJFHZM5EBWDREWAOZA5CNFSM4KYTPTJ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEM5OWQQ#issuecomment-591063874>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN4AB7MIANQIWQ5NPYOPLCLREWAOZANCNFSM4KYTPTJQ>
> .
>


-- 
Regards
Vadan

petere commented:
It sounds like your work is derived from their work, so just removing the license isn't going to work.

petere commented:
Sounds like this isn't going to work like this. So closing this PR for now.

xzilla commented:
For the record, the pgbouncer-rr license says:

```The terms reproduce, reproduction, derivative works, and distribution have the meaning as provided under U.S. copyright law; provided, however, that for the purposes of this License, derivative works shall not include works that remain separable from, or merely link (or bind by name) to the interfaces of, the Work.```

ISTM that the above work is separable from the original work, in case any one wants to pursue this further.

philippemnoel commented:
I might be interested in pursuing this further! I'll research the issue more, but adding query routing to pgbouncer would be very helpful for our use case

Vadangowda commented:
Go for it.

Regards
Vadan


On Thu, Jun 6, 2024 at 12:19 PM Philippe Noël ***@***.***>
wrote:

> I might be interested in pursuing this further! I'll research the issue
> more, but adding query routing to pgbouncer would be very helpful for our
> use case
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pgbouncer/pgbouncer/pull/466#issuecomment-2153245588>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN4AB7OEK37Z2CEDE2BN3ALZGCY3BAVCNFSM6AAAAABI5KTX6SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJTGI2DKNJYHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
